app/services/remove_status_service.rb

```diff
@@ -38,15 +40,48 @@ def remove_from_followers
     end
   end
 
-  def remove_from_mentioned(stream_entry)
-    salmon_xml       = stream_entry_to_xml(stream_entry)
-    target_accounts  = @mentions.map(&:account).reject(&:local?).uniq(&:domain)
+  def remove_from_remote_mentioned
+    # People who got mentioned in the status, or who
+    # reblogged it from someone else might not follow
+    # the author and wouldn't normally receive the
+    # delete notification - so here, we explicitly
+    # send it to them
+
+    target_accounts = (@mentions.map(&:account).reject(&:local?) + @reblogs.map(&:account).reject(&:local?)).uniq(&:id)
+
```

Comment:
If I have a status because someone I follow reblogged it, but I didn't reblog it myself, I wouldn't receive a delete. 

ActivityPub does not specify how to federate a Delete activity. tagging @cwebber because that feels like something the group should address.